vortex-array/src/builders/primitive.rs

```diff
@@ -184,57 +176,361 @@ impl<T: NativePType> ArrayBuilder for PrimitiveBuilder<T> {
 
 /// A range of uninitialized values in the primitive builder that can be filled.
 pub struct UninitRange<'a, T> {
-    offset: usize,
     len: usize,
     builder: &'a mut PrimitiveBuilder<T>,
 }
 
-impl<T> Deref for UninitRange<'_, T> {
```

Comment:
Yea this is the right move, these were added optimistically but were not useful after all, and indeed were wrong